File: app/src/main/java/com/yvrun/officeprocess/core/primary/home/HomeAdapter.java
package com.yvrun.officeprocess.core.primary.home;

import android.text.Html;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.yvrun.officeprocess.R;
import com.yvrun.officeprocess.global.GlideApp;
import com.yvrun.officeprocess.wight.CollectView;

public class HomeAdapter extends BaseQuickAdapter<HomeArticelResponseBean.DataBean.DatasBean, BaseViewHolder> {

    private OnCollectViewClickListener mOnCollectViewClickListener = null;

    public HomeAdapter() {
        super(R.layout.item_home);
    }

    public void setOnCollectViewClickListener(OnCollectViewClickListener onCollectViewClickListener) {
        mOnCollectViewClickListener = onCollectViewClickListener;
    }

    @Override
    protected void convert(BaseViewHolder helper, HomeArticelResponseBean.DataBean.DatasBean item) {
        helper.setText(R.id.tv_title, Html.fromHtml(item.getTitle()));
        helper.setText(R.id.tv_author, item.getAuthor());
        helper.setText(R.id.tv_time, item.getNiceDate());
        helper.setText(R.id.tv_super_chapter_name, item.getSuperChapterName());
        helper.setText(R.id.tv_chapter_name, item.getChapterName());
        LinearLayout ll_new = helper.getView(R.id.ll_new);
        if (item.isFresh()) {
            ll_new.setVisibility(View.VISIBLE);
        } else {
            ll_new.setVisibility(View.GONE);
        }
        ImageView iv_img = helper.getView(R.id.iv_img);
        if (!TextUtils.isEmpty(item.getEnvelopePic())) {
            GlideApp.with(iv_img.getContext()).load(item.getEnvelopePic()).into(iv_img);
            iv_img.setVisibility(View.VISIBLE);
        } else {
            iv_img.setVisibility(View.GONE);
        }
        CollectView cv_collect = helper.getView(R.id.cv_collect);
        if (item.isCollect()) {
            cv_collect.setChecked(true);
        } else {
            cv_collect.setChecked(false);
        }
        TextView tv_tag = helper.getView(R.id.tv_tag);
        if (item.getTags() != null && item.getTags().size() > 0) {
            tv_tag.setText(item.getTags().get(0).getName());
            tv_tag.setVisibility(View.VISIBLE);
        } else {
            tv_tag.setVisibility(View.GONE);
        }
        cv_collect.setOnClickListener(new CollectView.OnClickListener() {
            @Override
            public void onClick(CollectView v) {
                if (mOnCollectViewClickListener != null) {
                    mOnCollectViewClickListener.onClick(helper, v, helper.getAdapterPosition() - getHeaderLayoutCount());
                }
            }
        });
    }

    public interface OnCollectViewClickListener {
        void onClick(BaseViewHolder helper, CollectView v, int position);
    }
}

File: kotlinappother/src/main/java/com/chengzhen/kotlinappother/adapter/HomeAdapter.kt
package com.chengzhen.kotlinappother.adapter

import android.text.Html
import android.text.TextUtils
import android.view.View
import android.widget.CheckBox
import android.widget.CompoundButton
import android.widget.ImageView
import android.widget.TextView

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.chengzhen.kotlinappother.R
import com.chengzhen.kotlinappother.base.GlideApp
import com.chengzhen.kotlinappother.data.ArticleResponseBean

class HomeAdapter : BaseQuickAdapter<ArticleResponseBean.DataBean.DatasBean, BaseViewHolder>(R.layout.item_home) {

    private var mOnCollectViewClickListener: OnCollectViewClickListener? = null

    override fun convert(helper: BaseViewHolder, item: ArticleResponseBean.DataBean.DatasBean) {
        helper.setText(R.id.tv_title, Html.fromHtml(item.title))
                .setText(R.id.tv_author, item.author)
                .setText(R.id.tv_time, item.niceDate)
                .setText(R.id.tv_super_chapter_name, item.superChapterName)
                .setText(R.id.tv_chapter_name, item.chapterName)
                .setGone(R.id.ll_new, item.isFresh)

        val iv_img = helper.getView<ImageView>(R.id.iv_img)
        if (!TextUtils.isEmpty(item.envelopePic)) {
            GlideApp.with(iv_img.context).load(item.envelopePic).into(iv_img)
            iv_img.visibility = View.VISIBLE
        } else {
            iv_img.visibility = View.GONE
        }
        val cb_collect = helper.getView<CheckBox>(R.id.cb_collect)
        if (item.isCollect) {
            cb_collect.isChecked = true
        } else {
            cb_collect.isChecked = false
        }
        val tv_tag = helper.getView<TextView>(R.id.tv_tag)
        if (item.tags != null && item.tags.size > 0) {
            tv_tag.text = item.tags[0].name
            tv_tag.visibility = View.VISIBLE
        } else {
            tv_tag.visibility = View.GONE
        }

        cb_collect.setOnCheckedChangeListener { buttonView, isChecked ->
            if (mOnCollectViewClickListener != null) {
                mOnCollectViewClickListener!!.onClick(helper, buttonView, helper.adapterPosition - headerLayoutCount)
            }
        }
    }

    interface OnCollectViewClickListener {
        fun onClick(helper: BaseViewHolder, v: CompoundButton, position: Int)
    }

    fun setOnCollectViewClickListener(listener: OnCollectViewClickListener) {

        mOnCollectViewClickListener = listener
    }
}
